esphome/wizard.py

```
@@ -189,32 +191,49 @@ def wizard_write(path, **kwargs):
     from esphome.components.rtl87xx import boards as rtl87xx_boards
 
     name = kwargs["name"]
-    board = kwargs["board"]
-
-    for key in ("ssid", "psk", "password", "ota_password"):
-        if key in kwargs:
-            kwargs[key] = sanitize_double_quotes(kwargs[key])
-
-    if "platform" not in kwargs:
-        if board in esp8266_boards.BOARDS:
-            platform = "ESP8266"
-        elif board in esp32_boards.BOARDS:
-            platform = "ESP32"
-        elif board in rp2040_boards.BOARDS:
-            platform = "RP2040"
-        elif board in bk72xx_boards.BOARDS:
-            platform = "BK72XX"
-        elif board in ln882x_boards.BOARDS:
-            platform = "LN882X"
-        elif board in rtl87xx_boards.BOARDS:
-            platform = "RTL87XX"
-        else:
-            safe_print(color(AnsiFore.RED, f'The board "{board}" is unknown.'))
+    if kwargs["type"] == "empty":
+        file_text = ""
+        # Will be updated later after editing the file
+        hardware = "UNKNOWN"
+    elif kwargs["type"] == "upload":
+        try:
+            file_text = base64.b64decode(kwargs["file_content"]).decode("utf-8")
+        except (binascii.Error, UnicodeDecodeError):
```

This is failing on characters like `°`

---

We should probably also do this part in the webserver handler so we can propagate the error back to the user on the dashboard.

---

> This is failing on characters like `°`

I think you already fixed this with https://github.com/esphome/dashboard/pull/782/commits/78b223a2b247d5555a5fba422a0268ae7ae67520. At least it doesn't fail for me on files with ° and emojis anymore.

> We should probably also do this part in the webserver handler so we can propagate the error back to the user on the dashboard.

Done, also opened this PR:
- https://github.com/esphome/dashboard/pull/792

To show the specific error reason.